### ISSUE TITLE
Tyche 1.2.0 Incompatible with Atum 1.1.0 or greater

### ISSUE THREAD

**Issue by tildejustin** [VoidXWalker/tyche#1]
Tested in 1.16.1
```
MultiMC version: 0.7.0-develop-3524


Launched instance in online mode

authserver.mojang.com resolves to:
    [18.67.58.61, 18.67.58.60, 18.67.58.35, 18.67.58.44]

session.minecraft.net resolves to:
    [100.24.172.95, 34.199.236.154, 35.153.153.200]

textures.minecraft.net resolves to:
    [18.67.65.82, 18.67.65.91, 18.67.65.46, 18.67.65.85]

api.mojang.com resolves to:
    [18.160.37.118, 18.160.37.66, 18.160.37.115, 18.160.37.49]


Minecraft folder is:
D:/Games/Minecraft/MultiMC/instances/1.16.1_Debug/.minecraft


Java path is:
D:/Programs/Program Files/Java/Adoptium 17/bin/javaw.exe


Checking Java version...
Java is version 17.0.4.1, using 64-bit architecture, from Eclipse Adoptium.


Main Class:
  net.fabricmc.loader.impl.launch.knot.KnotClient

Native path:
  D:/Games/Minecraft/MultiMC/instances/1.16.1_Debug/natives

Traits:
traits FirstThreadOnMacOS

Libraries:
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl-glfw/3.2.2/lwjgl-glfw-3.2.2.jar
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl-jemalloc/3.2.2/lwjgl-jemalloc-3.2.2.jar
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl-openal/3.2.2/lwjgl-openal-3.2.2.jar
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl-opengl/3.2.2/lwjgl-opengl-3.2.2.jar
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl-stb/3.2.2/lwjgl-stb-3.2.2.jar
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl-tinyfd/3.2.2/lwjgl-tinyfd-3.2.2.jar
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl/3.2.2/lwjgl-3.2.2.jar
  D:/Games/Minecraft/MultiMC/libraries/com/mojang/patchy/1.3.9/patchy-1.3.9.jar
  D:/Games/Minecraft/MultiMC/libraries/oshi-project/oshi-core/1.1/oshi-core-1.1.jar
  D:/Games/Minecraft/MultiMC/libraries/net/java/dev/jna/jna/4.4.0/jna-4.4.0.jar
  D:/Games/Minecraft/MultiMC/libraries/net/java/dev/jna/platform/3.4.0/platform-3.4.0.jar
  D:/Games/Minecraft/MultiMC/libraries/com/ibm/icu/icu4j/66.1/icu4j-66.1.jar
  D:/Games/Minecraft/MultiMC/libraries/com/mojang/javabridge/1.0.22/javabridge-1.0.22.jar
  D:/Games/Minecraft/MultiMC/libraries/net/sf/jopt-simple/jopt-simple/5.0.3/jopt-simple-5.0.3.jar
  D:/Games/Minecraft/MultiMC/libraries/io/netty/netty-all/4.1.25.Final/netty-all-4.1.25.Final.jar
  D:/Games/Minecraft/MultiMC/libraries/com/google/guava/guava/21.0/guava-21.0.jar
  D:/Games/Minecraft/MultiMC/libraries/org/apache/commons/commons-lang3/3.5/commons-lang3-3.5.jar
  D:/Games/Minecraft/MultiMC/libraries/commons-io/commons-io/2.5/commons-io-2.5.jar
  D:/Games/Minecraft/MultiMC/libraries/commons-codec/commons-codec/1.10/commons-codec-1.10.jar
  D:/Games/Minecraft/MultiMC/libraries/com/mojang/brigadier/1.0.17/brigadier-1.0.17.jar
  D:/Games/Minecraft/MultiMC/libraries/com/mojang/datafixerupper/3.0.25/datafixerupper-3.0.25.jar
  D:/Games/Minecraft/MultiMC/libraries/com/google/code/gson/gson/2.8.0/gson-2.8.0.jar
  D:/Games/Minecraft/MultiMC/libraries/com/mojang/authlib/1.6.25/authlib-1.6.25.jar
  D:/Games/Minecraft/MultiMC/libraries/org/apache/commons/commons-compress/1.8.1/commons-compress-1.8.1.jar
  D:/Games/Minecraft/MultiMC/libraries/org/apache/httpcomponents/httpclient/4.3.3/httpclient-4.3.3.jar
  D:/Games/Minecraft/MultiMC/libraries/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar
  D:/Games/Minecraft/MultiMC/libraries/org/apache/httpcomponents/httpcore/4.3.2/httpcore-4.3.2.jar
  D:/Games/Minecraft/MultiMC/libraries/it/unimi/dsi/fastutil/8.2.1/fastutil-8.2.1.jar
  D:/Games/Minecraft/MultiMC/libraries/org/apache/logging/log4j/log4j-api/2.16.0/log4j-api-2.16.0.jar
  D:/Games/Minecraft/MultiMC/libraries/org/apache/logging/log4j/log4j-core/2.16.0/log4j-core-2.16.0.jar
  D:/Games/Minecraft/MultiMC/libraries/com/mojang/text2speech/1.11.3/text2speech-1.11.3.jar
  D:/Games/Minecraft/MultiMC/libraries/net/fabricmc/intermediary/1.16.1/intermediary-1.16.1.jar
  D:/Games/Minecraft/MultiMC/libraries/net/fabricmc/tiny-mappings-parser/0.3.0+build.17/tiny-mappings-parser-0.3.0+build.17.jar
  D:/Games/Minecraft/MultiMC/libraries/net/fabricmc/sponge-mixin/0.11.4+mixin.0.8.5/sponge-mixin-0.11.4+mixin.0.8.5.jar
  D:/Games/Minecraft/MultiMC/libraries/net/fabricmc/tiny-remapper/0.8.2/tiny-remapper-0.8.2.jar
  D:/Games/Minecraft/MultiMC/libraries/net/fabricmc/access-widener/2.1.0/access-widener-2.1.0.jar
  D:/Games/Minecraft/MultiMC/libraries/org/ow2/asm/asm/9.3/asm-9.3.jar
  D:/Games/Minecraft/MultiMC/libraries/org/ow2/asm/asm-analysis/9.3/asm-analysis-9.3.jar
  D:/Games/Minecraft/MultiMC/libraries/org/ow2/asm/asm-commons/9.3/asm-commons-9.3.jar
  D:/Games/Minecraft/MultiMC/libraries/org/ow2/asm/asm-tree/9.3/asm-tree-9.3.jar
  D:/Games/Minecraft/MultiMC/libraries/org/ow2/asm/asm-util/9.3/asm-util-9.3.jar
  D:/Games/Minecraft/MultiMC/libraries/net/fabricmc/fabric-loader/0.14.8/fabric-loader-0.14.8.jar
  D:/Games/Minecraft/MultiMC/libraries/com/mojang/minecraft/1.16.1/minecraft-1.16.1-client.jar

Native libraries:
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl-glfw/3.2.2/lwjgl-glfw-3.2.2-natives-windows.jar
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl-jemalloc/3.2.2/lwjgl-jemalloc-3.2.2-natives-windows.jar
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl-openal/3.2.2/lwjgl-openal-3.2.2-natives-windows.jar
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl-opengl/3.2.2/lwjgl-opengl-3.2.2-natives-windows.jar
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl-stb/3.2.2/lwjgl-stb-3.2.2-natives-windows.jar
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl-tinyfd/3.2.2/lwjgl-tinyfd-3.2.2-natives-windows.jar
  D:/Games/Minecraft/MultiMC/libraries/org/lwjgl/lwjgl/3.2.2/lwjgl-3.2.2-natives-windows.jar
  D:/Games/Minecraft/MultiMC/libraries/com/mojang/text2speech/1.11.3/text2speech-1.11.3-natives-windows.jar

Mods:
  [❌] antiresourcereload-1.2.1+1.16.1.jar.disabled (disabled)
  [❌] atum-1.0.5+1.16.1.jar.disabled (disabled)
  [✔️] atum-1.1.0+1.16.1.jar
  [❌] atum-1.1.2+1.16.1.jar.disabled (disabled)
  [❌] BiomeThreadLocalFix-1.3.jar.disabled (disabled)
  [❌] chunkmod-1.1.0.jar.disabled (disabled)
  [❌] extra-options-1.0.0-1.16.jar.disabled (disabled)
  [❌] fast-reset-1.16.1-v1.4.1.jar.disabled (disabled)
  [❌] forceport-1.1.0.jar.disabled (disabled)
  [❌] krypton-1.16.1-backport-0.1.3-SNAPSHOT+2021-02-20.jar.disabled (disabled)
  [❌] lazydfu-0.1.2.jar.disabled (disabled)
  [❌] lazystronghold-1.1.3+1.16.1.jar.disabled (disabled)
  [❌] lithium-1.16.1-backport-fabric-0.6.6.jar.disabled (disabled)
  [❌] no_peaceful-1.0.1.jar.disabled (disabled)
  [📁] rong (folder)
  [❌] sleepbackground-3.7.1-1.15.x-1.19.x.jar.disabled (disabled)
  [❌] sodium-1.16.1-backport-fabric-0.2.1+build.10.jar.disabled (disabled)
  [❌] SpeedRunIGT-11.1+1.16.1.jar.disabled (disabled)
  [❌] standardsettings-1.1.2+1.16.1.jar.disabled (disabled)
  [❌] standardsettings-1.2.1.1+1.16.1.jar.disabled (disabled)
  [❌] standardsettings-1.2.2+1.16.1.jar.disabled (disabled)
  [❌] starlight-fabric-1.0.0-RC2-1.16.x.jar.disabled (disabled)
  [✔️] tyche-1.2.0+1.16.1.jar
  [❌] tyche-1.2.0+1.16.5.jar.disabled (disabled)
  [❌] voyager-1.0.0.jar.disabled (disabled)
  [❌] worldpreview-2.3.3+1.16.1.jar.disabled (disabled)

Params:
  --username  --version 1.16.1 --gameDir D:/Games/Minecraft/MultiMC/instances/1.16.1_Debug/.minecraft --assetsDir D:/Games/Minecraft/MultiMC/assets --assetIndex 1.16 --uuid  --accessToken  --userType  --versionType release

Window size: 854 x 480

Java Arguments:
[-XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump, -Xms1024m, -Xmx2048m, -Duser.language=en]


Minecraft process ID: 6168


Using onesix launcher.

[18:14:57] [main/INFO]: Loading Minecraft 1.16.1 with Fabric Loader 0.14.8
[18:14:58] [main/INFO]: Loading 5 mods:
	- atum 1.1.0+1.16.1
	- fabricloader 0.14.8
	- java 17
	- minecraft 1.16.1
	- tyche 1.2.0+1.16.1
[18:14:58] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/D:/Games/Minecraft/MultiMC/libraries/net/fabricmc/sponge-mixin/0.11.4+mixin.0.8.5/sponge-mixin-0.11.4+mixin.0.8.5.jar Service=Knot/Fabric Env=CLIENT
[18:14:58] [main/INFO]: Compatibility level set to JAVA_16
[18:15:05] [Render thread/WARN]: Method overwrite conflict for ping in tyche.mixins.json:LevelLoadingScreenMixin from mod tyche, previously written by me.voidxwalker.autoreset.mixin.LevelLoadingScreenMixin. Skipping method.
[18:15:05] [Render thread/WARN]: Injection warning: LVT in net/minecraft/class_3928::method_25394(Lnet/minecraft/class_4587;IIF)V has incompatible changes at opcode 119 in callback tyche.mixins.json:LevelLoadingScreenMixin from mod tyche->@Inject::modifyString(Lnet/minecraft/class_4587;IIFLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;Ljava/lang/String;II)V.
 Expected: [Ljava/lang/String;, I, I]
    Found: [Ljava/lang/String;, J, I]
Available: [Ljava/lang/String;, J, I, I, I]
[18:15:05] [Render thread/WARN]: Method overwrite conflict for ping in tyche.mixins.json:LevelLoadingScreenMixin from mod tyche, previously written by me.voidxwalker.autoreset.mixin.LevelLoadingScreenMixin. Skipping method.
[18:15:05] [Render thread/WARN]: Injection warning: LVT in net/minecraft/class_3928::method_25394(Lnet/minecraft/class_4587;IIF)V has incompatible changes at opcode 119 in callback tyche.mixins.json:LevelLoadingScreenMixin from mod tyche->@Inject::modifyString(Lnet/minecraft/class_4587;IIFLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;Ljava/lang/String;II)V.
 Expected: [Ljava/lang/String;, I, I]
    Found: [Ljava/lang/String;, J, I]
Available: [Ljava/lang/String;, J, I, I, I]
[18:15:05] [Render thread/ERROR]: Minecraft has crashed!
net.fabricmc.loader.impl.FormattedException: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_3928 failed
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:463) ~[fabric-loader-0.14.8.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.8.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.8.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:214) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:249) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_3928 failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:419) ~[fabric-loader-0.14.8.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.8.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.8.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.8.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.client.main.Main.main(Main.java:185) ~[minecraft-1.16.1-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.8.jar:?]
	... 10 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:414) ~[fabric-loader-0.14.8.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.8.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.8.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.8.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.client.main.Main.main(Main.java:185) ~[minecraft-1.16.1-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.8.jar:?]
	... 10 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Callback method modifyString(Lnet/minecraft/class_4587;IIFLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;Ljava/lang/String;II)V in tyche.mixins.json:LevelLoadingScreenMixin from mod tyche failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap tyche-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1385) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1052) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:414) ~[fabric-loader-0.14.8.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.8.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.8.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.8.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.client.main.Main.main(Main.java:185) ~[minecraft-1.16.1-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.8.jar:?]
	... 10 more
Process exited with code 1 (0x1).
Please note that usually neither the exit code, nor its description are enough to diagnose issues!
Always upload the entire log and not just the exit code.
Clipboard copy at: 21 Sep 2022 18:15:07 -0400
```

**Comment by KavxKo0L** [comment]
yes i have the same error

**Comment by urntt** [comment]
same

**Comment by tildejustin** [comment]
fixed as of tyche 1.2.2
